chain/client/src/client.rs

```diff
@@ -1083,6 +1083,11 @@ impl Client {
 
     /// If we are close to epoch boundary, return next epoch id, otherwise return None.
     fn get_next_epoch_id_if_at_boundary(&self, head: &Tip) -> Result<Option<EpochId>, Error> {
+        let next_epoch_started =
+            self.runtime_adapter.is_next_block_epoch_start(&head.last_block_hash)?;
+        if next_epoch_started {
+            return Ok(None);
```

Comment:
Actually what is this special case? if next block is the start of the epoch why do we not return next epoch id?

Reply:
Because we're not at epoch boundary.
`head` is previous block. That epoch is over, for the near future all blocks are in the same epoch

Reply:
`epoch_id` is defined this way: 
```
let epoch_id = self.runtime_adapter.get_epoch_id_from_prev_block(&head.last_block_hash)?;
```
next epoch id is the epoch after `epoch_id`. It's far away.

Reply:
Ah right